[benjamin-wee] iP

benjamin-wee opened the pull request:
## DukeMaster
> "Productivity is never an accident. It is always the result of a commitment to excellence, intelligent planning, and focused effort." - Paul J. Meyer [(source)](https://www.brainyquote.com/quotes/paul_j_meyer_393225) 

DukeMaster helps you keep track of your tasks, so you can finally hit your deadlines and reach your goals. It is:
+ intuitive 
+ command-line friendly
+ ~~very~~ **ULTRA** simple to use 

You're *this* close to getting your life together!

You simply have to:

1. Download DukeMaster [here](https://github.com/benjamin-wee/ip/releases/tag/A-Jar)
2. Start up DukeMaster
3. Log your tasks
4. And that's it!

You won't have to worry about forgetting tasks anymore! 🤩 

Features:

- [ ] Manage deadlines
- [ ] Manage events
- [ ] Manage ToDos

If you are a Java Programmer, it's easier than printing "Hello World" with the `main` method shown below!
```Java
public class Main {
    public static void main(String[] args) {
        System.out.println("Hello World");
    }
}
```

Commit:
Add Gradle support

benjamin-wee committed:
first commit

benjamin-wee committed:
Second commit

benjamin-wee committed:
added marking and unmarking status of tasks

benjamin-wee committed:
added deadline, event, todo classes

benjamin-wee committed:
added automated testing of text UIs and minor edits to duke.java

benjamin-wee committed:
handle exceptions

benjamin-wee committed:
added delete command

benjamin-wee committed:
added save feature and dates

benjamin-wee committed:
Revert "added save feature and dates"

This reverts commit 7d90cdfb7b2704c6b03cdfe2169354ccbf68b21b.

benjamin-wee committed:
no message

benjamin-wee committed:
added saving to hard disk

benjamin-wee committed:
allowed Duke to understand dates

benjamin-wee committed:
no message

benjamin-wee committed:
Merge branch 'branch-Level-8'

* branch-Level-8:
  no message
  allowed Duke to understand dates

wongyx commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,215 @@
+import java.util.*;
```
Perhaps it would be better to list out the specific imports that you need from java.util

wongyx commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,215 @@
+                        list.add(todo);
+                        System.out.println("Got it. I've added this task:\n" + todo.toString() + "\nNow you have "
+                                + list.size() + " tasks in the list.");
+                    } else if (input.contains("deadline")) {
```
Not sure if my logic is wrong but would it be better to check if input starts with deadline instead of contain? If somehow the input is "deadline make a todo app /by 2023/02/01" does it cause the code to excercute the previous if statement block of todo?

```suggestion
                    } else if (input.startsWith("deadline")) {

wongyx reviewed (commented):
There might be some minor issues with the code but overall it is great!!

benjamin-wee committed:
added classes TaskList, Ui, Parser, Storage

junlee1991 commented on src/main/java/Storage.java:
```diff
@@ -0,0 +1,98 @@
+import java.util.Scanner;
+import java.util.ArrayList;
+import java.io.File;
+import java.io.FileNotFoundException;
```
The ordering of import statements must be consistent. (alphabetically)

junlee1991 commented on src/main/java/TaskList.java:
```diff
@@ -0,0 +1,120 @@
+                LocalDate dateString = LocalDate.parse(date);
+                Deadline deadline = new Deadline(description, dateString);
+                taskList.add(deadline);
+                System.out.println("Got it. I've added this task:\n" + deadline.toString() + "\nNow you have "
```
Should put this in the UI since it has to do with printing

junlee1991 commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,49 @@
+import java.util.Scanner;
+
 public class Duke {
```
Overall, I found your code easy to read and understand!

benjamin-wee committed:
organised classes into duke package

benjamin-wee committed:
Merge remote-tracking branch 'upstream/add-gradle-support'
# Please enter a commit message to explain why this merge is necessary,
# especially if it merges an updated upstream into a topic branch.
#
# Lines starting with '#' will be ignored, and an empty message aborts
# the commit.
set up Gradle

junlee1991 reviewed (commented):
Good Job! keep it up!

benjamin-wee committed:
added JUnit tests

jednhk commented on src/main/java/duke/Task.java:
```diff
@@ -0,0 +1,40 @@
+    protected Boolean isDone;
+
+    public Task(String description) throws MissingDescriptionException {
+        if (description.equals("") || description.equals(" ")) {
```
Do you need to consider the case where the description is two spaces or more?

bryanongjx replied:
can consider using isEmptyString instead !

jednhk commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,125 @@
+            try {
+                Todo todo = new Todo(taskDescription);
+                taskList.add(todo);
+                System.out.println("Got it. I've added this task:\n" + todo.toString() + "\nNow you have "
```
Perhaps move this to the UI class? Since it has to do with user interaction

jednhk reviewed (commented):
LGTM. Perhaps will be easier to debug in the future if the print statements are in the Ui class instead.

bryanongjx commented on src/main/java/duke/MissingDescriptionException.java:
```diff
@@ -0,0 +1,13 @@
+package duke;
```
good idea creating a missingdescriptionexception class!

bryanongjx commented on src/main/java/duke/Task.java:
```diff
@@ -0,0 +1,40 @@
+package duke;
+
+public class Task {
```
can consider making this class abstract! not sure though, but worth a thought!

bryanongjx commented on src/main/java/duke/Deadline.java:
```diff
@@ -0,0 +1,29 @@
+package duke;
+
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+
+public class Deadline extends Task {
+    private static final String taskType = "[D]";
+    private LocalDate date;
+
+    public Deadline(String description, LocalDate date) throws MissingDescriptionException {
+        super(description);
+        this.date = date;
+    }
+
+    @Override
+    public String toString() {
+        DateTimeFormatter dateFormatter = DateTimeFormatter.ofPattern("d MMM u");
+        String dateString = date.format(dateFormatter);
+        return taskType + super.toString() + " (by: " + dateString + ")";
+    }
+
+    @Override
+    public String toStorageData() {
+        String completed = this.getStatusIcon();
+        DateTimeFormatter dateFormatter = DateTimeFormatter.ofPattern("d MMM u");
+        String dateString = date.format(dateFormatter);
+        return taskType + "//" + completed + "//" + description + "//" + dateString;
+    }
+}
```
can consider adding documentation for your classes

benjamin-wee committed:
added JavaDocs

benjamin-wee committed:
edited JavaDocs

benjamin-wee committed:
edited code to comply to coding standards

benjamin-wee committed:
added feature to find tasks using a keyword

benjamin-wee committed:
Merge branch 'branch-A-CodingStandard'

* branch-A-CodingStandard:
  edited code to comply to coding standards

# Conflicts:
#	src/main/java/duke/Parser.java
#	src/main/java/duke/Storage.java
#	src/main/java/duke/TaskList.java

benjamin-wee committed:
Merge branch 'branch-Level-9'

* branch-Level-9:
  added feature to find tasks using a keyword

# Conflicts:
#	data/duke.txt
#	src/main/java/duke/TaskList.java

benjamin-wee committed:
updated JavaDocs for findTask method

benjamin-wee committed:
added GUI

benjamin-wee committed:
add stylesheet

benjamin-wee committed:
add assertions

benjamin-wee committed:
no message

benjamin-wee committed:
Improve the code quality of the code

There are several methods that are longer than 30 lines of code, poor
SLAP, use magic strings, and where happy paths are not prominent

Improving code quality improves understandability and makes it easier to
debug

As a step towards improving code quality, let's extract out parts of the
code into separate methods in their respective classes, to adhere to
SLAP and make the methods lesser than 30 lines. Let's also remove the
use of magic literals by using a named constant

benjamin-wee committed:
Merge pull request #2 from benjamin-wee/branch-A-Assertions

add assertions

benjamin-wee committed:
Merge branch 'master' into branch-A-CodeQuality

benjamin-wee committed:
Merge pull request #3 from benjamin-wee/branch-A-CodeQuality

Improve the code quality of the code

benjamin-wee committed:
bug fix

benjamin-wee committed:
add extension D-Notes

benjamin-wee committed:
update javadocs

benjamin-wee committed:
improve GUI

benjamin-wee committed:
fix bugs

benjamin-wee committed:
change indexOutOfBoundsException error message

benjamin-wee committed:
add Ui.png and brief User Guide(UG)

benjamin-wee committed:
change mainClassName under application

benjamin-wee committed:
update TaskList JUnit test

benjamin-wee committed:
fix bug and update User Guide
